plugins/by-name/trouble/default.nix

```diff
@@ -115,200 +115,248 @@ helpers.neovim-plugin.mkNeovimPlugin {
+                 view:filter({ severity = severity }, {
+                   id = "severity",
+                   template = "{hl:Title}Filter:{hl} {severity}",
+                   del = severity == 0,
+                 })
+              end
+            '';
+            desc = "Toggle Severity Filter";
+          };
+        }
+        ''
+          Key mappings can be set to the name of a builtin action,
+          or you can define your own custom action.
+        '';
 
-    use_diagnostic_signs = helpers.defaultNullOpts.mkBool false ''
-      Enabling this will use the signs defined in your lsp client
-    '';
+    modes =
+      defaultNullOpts.mkAttrsOf lib.types.anything
+        {
+          lsp_references = {
+            params = {
+              include_declaration = true;
+            };
+          };
+          lsp_base = {
+            params = {
+              include_current = false;
+            };
+          };
+          symbols = {
+            desc = "document symbols";
+            mode = "lsp_document_symbols";
+            focus = false;
+            win = {
+              position = "right";
+            };
+            filter = {
+              "not" = {
+                ft = "lua";
+                kind = "Package";
+              };
+              any = {
+                ft = [
+                  "help"
+                  "markdown"
+                ];
+                kind = [
+                  "Class"
+                  "Constructor"
+                  "Enum"
+                  "Field"
+                  "Function"
+                  "Interface"
+                  "Method"
+                  "Module"
+                  "Namespace"
+                  "Package"
+                  "Property"
+                  "Struct"
+                  "Trait"
+                ];
+              };
+            };
+          };
+        }
+        ''
+          Customize the various builtin modes or define your own.
+        '';
+
+    icons = defaultNullOpts.mkAttrsOf lib.types.anything {
+      indent = {
+        top = "│ ";
+        middle = "├╴";
+        last = "└╴";
+        fold_open = " ";
+        fold_closed = " ";
+        ws = "  ";
+      };
+      folder_closed = " ";
+      folder_open = " ";
+      kinds = {
+        Array = " ";
+        Boolean = "󰨙 ";
+        Class = " ";
+        Constant = "󰏿 ";
+        Constructor = " ";
+        Enum = " ";
+        EnumMember = " ";
+        Event = " ";
+        Field = " ";
+        File = " ";
+        Function = "󰊕 ";
+        Interface = " ";
+        Key = " ";
+        Method = "󰊕 ";
+        Module = " ";
+        Namespace = "󰦮 ";
+        Null = " ";
+        Number = "󰎠 ";
+        Object = " ";
+        Operator = " ";
+        Package = " ";
+        Property = " ";
+        String = " ";
+        Struct = "󰆼 ";
+        TypeParameter = " ";
+        Variable = "󰀫 ";
+      };
+    } "Define custom icons.";
   };
 
   extraConfig = cfg: {
```

Comment:
Do you want to warn when the old settings attrs are present in `cfg.settings`?

Reply:
Thinking just a warning about deprecation/removal or mapping them to a mkRemovedOption for an actual assertion?

Reply:
I don't think renames are appropriate for freeform settings options.

I'm fine with doing nothing, but if you want to do something I'd suggest just a warning if x, y, or z attrs are present under `cfg.settings`.

I think you did something similar in another PR recently?

Reply:
> I don't think renames are appropriate for freeform settings options.
> 
> I'm fine with doing nothing, but if you want to do something I'd suggest just a warning if x, y, or z attrs are present under `cfg.settings`.
> 
> I think you did something similar in another PR recently?

Yeah, looks like it was the rest.nvim v3 migration, I can do that again probably

Reply:
Heading out the door for a little family time for a bit, I pushed a quick implementation but I'll double check the list of fields when I get back later before merge.